h2/src/main/org/h2/util/DateTimeUtils.java

```diff
@@ -627,8 +604,23 @@ public static int getDatePart(Value date, int field) {
             return (int) (timeNanos / 1_000_000_000 % 60);
         case Calendar.MILLISECOND:
             return (int) (timeNanos / 1_000_000 % 1_000);
+        case Calendar.DAY_OF_YEAR:
+            return getDayOfYear(dateValue);
+        case Calendar.DAY_OF_WEEK:
+            return getSundayDayOfWeek(dateValue);
+        case Calendar.WEEK_OF_YEAR:
+            GregorianCalendar gc = getCalendar();
+            return getWeekOfYear(dateValue, gc.getFirstDayOfWeek() - 1, gc.getMinimalDaysInFirstWeek());
+        case Function.QUARTER:
```

Comment:
passing constants from two different classes in the same field is a little dubious i.e Function and Calendar, rather split this method

Reply:
I replaced all constants with constants from `Function`. This produces some more lines in `dateadd()`, but this allows to merge cases in `getSimpleValue()` that had a larger code.

Reply:
I initially planned to use some common constants, but forgot about it later.

I think that we don't need a new constants or new enumeration for this case, Function already have almost all of them except for `MILLISECOND`. Now this new constant is not mapped to any normal function, but if such function will be required in can be used for it, some databases have more separate functions like `MICROSECOND()` etc.